mylar/webserve.py

```diff
@@ -4924,6 +4924,83 @@ def importReadlist(self, filename):
         raise cherrypy.HTTPRedirect("detailStoryArc?StoryArcID=%s&StoryArcName=%s" % (storyarcid, storyarc))
     importReadlist.exposed = True
 
+    def GenerateCBLReadingList(self, StoryArcID=None):
+
+        myDB = db.DBConnection()
+
+        if StoryArcID:
+            ReadingArc = myDB.select("SELECT * FROM storyarcs WHERE StoryArcID=? AND (Manual IS NOT 'deleted' OR Manual IS NULL) ORDER BY ReadingOrder", [StoryArcID])
+
+            if ReadingArc is None:
+                logger.info("Couldn't find Story Arc to use for reading list!")
+            else:
+                arcName = ReadingArc[0]['StoryArc']
+                logger.fdebug('Generating CBL file for ' + arcName)
+
+                # !Replace root_path variable
+                readingListDest = os.path.join(mylar.CONFIG.DESTINATION_DIR, 'ReadingLists')
+                filename = arcName + ".cbl"
+                fullPath = os.path.join(readingListDest, filename)
+
+                if not os.path.exists(readingListDest):
+                    logger.fdebug('making directory: %s' % readingListDest)
+                    filechecker.validateAndCreateDirectory(readingListDest, True)
+
+                logger.fdebug('Opening file: ' + fullPath)
+
+                with open(fullPath, 'w') as file:
+
+                    line1 = "<?xml version=\"1.0\"?>"
+                    line2 = "<ReadingList xmlns:xsd=\"http://www.w3.org/2001/XMLSchema\" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\">"
+                    line3 = "<Name>" + escape(arcName) + "</Name>"
+                    line4 = "<Books>"
+
+                    file.writelines([line1, line2, line3, line4])
+
+                    #For each issue in arc
+                    for issue in ReadingArc:
+
+                        #Set default values to blank string
+                        fileName = seriesName = issueNumber = volNumber = issueYear = ""
+
+                        if issue['Location'] is not None:
+                            filePath = issue['Location'].split(os.path.sep)
+                            #lastItemIndex = len(filePath) - 1
+                            fileName = escape(filePath[-1])
+                            
+                            #Strip reading order numbers from start of filename if present
+                            regex = re.compile("^\d{3}-(.*?)\.(cbz|cbr)")
+                            match = regex.match(fileName)
+                            if match != None:
+                                strippedName = '{}.{}'.format(*match.groups())
+                                fileName=strippedName
+
+                        if issue['ComicName'] is not None:
+                            seriesName = escape(str(issue['ComicName']))
+                        if issue['IssueNumber'] is not None:
+                            issueNumber = escape(str(issue['IssueNumber']))
```

Comment:
escape isn't needed - just assign it as the value without the string (you could keep it, but it's not needed).